src/coordinatorapi.cpp

```diff
@@ -78,24 +78,24 @@ static int _cachedPort = 0;
 
 void
 CoordinatorAPI::getCoordHostAndPort(CoordinatorMode mode,
-                                    const char **host,
+                                    string &host,
                                     int *port)
 {
   if (SharedData::initialized()) {
-    *host = SharedData::coordHost().c_str();
```

Comment:
This is a use-after-free error.

`getCoordHostAndPort()` gets a copy of the string returned by `coordHost()`. We save a pointer to this string in `*host`. However, the string object is de-allocated as soon as we return from `getCoordHostAndPort()`.